README.md

```diff
@@ -1,12 +1,33 @@
 # plearn
+What is this?
+
+Explanation 1:
+
+  Duolingo for programming languages.
+
+Explanation 2 (Algorithmically Selected "Chapter Exercises"):
+
+  You're studying a textbook (do those still exist?).  At the end of the
+  chapter are problems that help you to:
+
+    * practice knowledge skills
+    * learn what you don't know
+    * learn what you do know
+
+ This tool uses information about you (bound to a pseudonym), primarily the 
+ history of your exercise responses, and others who are interested in the same
+ topic to guess which _next_ exercise will teach you the most.
```

Comment:
Just one more note: we could measure "elegance" by how many lines of code were required for a correct solution.  Of course that would encourage players to use "one-liners", so maybe factor in number of characters per line as well.

Abstractness could be measured by throwing lots of different types of objects at a potential solution and see if they are handled correctly, or at least sensibly.

Efficiency could obviously be measured by time/space complexity.